trace trap error on macOS

We have noticed that fits2idia no longer works on any version of macOS. The problem is easy to reproduce.  Here is an example:
```
git clone https://github.com/CARTAvis/fits2idia.git
cd fits2idia
mkdir build
cd build
cmake ..
make
./fits2idia ~/HD163296_CO_2_1.fits 
zsh: trace trap  ./fits2idia ~/HD163296_CO_2_1.fits
```

However, we found that fits2idia installed by Homebrew works fine on the same macOS computer. 
The key is that the Homebrew formula downloads a tarball of the v0.1.15 source code from Jun 1 2022: https://github.com/CARTAvis/fits2idia/archive/refs/tags/v0.1.15.tar.gz

If the source code from that Jun 1 2022 tarball is built and run on macOS, then fits2idia works perfectly without errors: e.g.
```
wget https://github.com/CARTAvis/fits2idia/archive/refs/tags/v0.1.15.tar.gz
tar -xvf v0.1.15.tar.gz
cd fits2idia-0.1.15
mkdir build
cd build
cmake ..
make
 ./fits2idia ~/HD163296_CO_2_1.fits 
```

Therefore, it appears something was introduced into the fits2idia 'dev' branch between Jun 1 2022 and today that has broken it on macOS. 

I am only reporting the issue here. I have not checked back through the commits to find out exactly when the problem was introduced.

In future, it would be a good idea to add some CI to fits2idia.

@confluence Indeed, that is where the problem begins!

Commit 968617f gives the "trace trap" error but the commit before it, 67d82e2e, gives no error. 

If it is any help, we can see some warnings in the "make" output;

The working `67d82e2e` commit output is: 
```
[ 12%] Building CXX object CMakeFiles/fits2idia.dir/main.cc.o
[ 25%] Building CXX object CMakeFiles/fits2idia.dir/Stats.cc.o
[ 37%] Building CXX object CMakeFiles/fits2idia.dir/MipMap.cc.o
[ 50%] Building CXX object CMakeFiles/fits2idia.dir/Converter.cc.o
/Users/acdc/fits2idia/Converter.cc:69:1: warning: non-void function does not return a value [-Wreturn-type]
}
^
1 warning generated.
[ 62%] Building CXX object CMakeFiles/fits2idia.dir/FastConverter.cc.o
[ 75%] Building CXX object CMakeFiles/fits2idia.dir/SlowConverter.cc.o
[ 87%] Building CXX object CMakeFiles/fits2idia.dir/Util.cc.o
[100%] Linking CXX executable fits2idia
[100%] Built target fits2idia
```

The non-working  `968617f` make output is:
```
[ 12%] Building CXX object CMakeFiles/fits2idia.dir/main.cc.o
In file included from /Users/acdc/fits2idia/main.cc:7:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/regex:768:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__locale:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/string:522:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/algorithm:653:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/functional:500:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__functional/function.h:20:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:25:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/unique_ptr.h:57:5: warning: delete called on 'Converter' that is abstract but has non-virtual destructor [-Wdelete-abstract-non-virtual-dtor]
    delete __ptr;
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/unique_ptr.h:318:7: note: in instantiation of member function 'std::default_delete<Converter>::operator()' requested here
      __ptr_.second()(__tmp);
      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/unique_ptr.h:272:19: note: in instantiation of member function 'std::unique_ptr<Converter>::reset' requested here
  ~unique_ptr() { reset(); }
                  ^
/Users/acdc/fits2idia/main.cc:121:32: note: in instantiation of member function 'std::unique_ptr<Converter>::~unique_ptr' requested here
    std::unique_ptr<Converter> converter;
                               ^
1 warning generated.
[ 25%] Building CXX object CMakeFiles/fits2idia.dir/Stats.cc.o
[ 37%] Building CXX object CMakeFiles/fits2idia.dir/MipMap.cc.o
[ 50%] Building CXX object CMakeFiles/fits2idia.dir/Converter.cc.o
In file included from /Users/acdc/fits2idia/Converter.cc:6:
In file included from /Users/acdc/fits2idia/Converter.h:9:
In file included from /Users/acdc/fits2idia/common.h:9:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/vector:275:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__bit_reference:15:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/algorithm:653:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/functional:500:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__functional/function.h:20:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:25:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/unique_ptr.h:57:5: warning: delete called on 'Converter' that is abstract but has non-virtual destructor [-Wdelete-abstract-non-virtual-dtor]
    delete __ptr;
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/unique_ptr.h:318:7: note: in instantiation of member function 'std::default_delete<Converter>::operator()' requested here
      __ptr_.second()(__tmp);
      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/unique_ptr.h:272:19: note: in instantiation of member function 'std::unique_ptr<Converter>::reset' requested here
  ~unique_ptr() { reset(); }
                  ^
/Users/acdc/fits2idia/Converter.cc:57:16: note: in instantiation of member function 'std::unique_ptr<Converter>::~unique_ptr' requested here
        return std::unique_ptr<Converter>(new SlowConverter(inputFileName, outputFileName, progress));
               ^
1 warning generated.
[ 62%] Building CXX object CMakeFiles/fits2idia.dir/FastConverter.cc.o
[ 75%] Building CXX object CMakeFiles/fits2idia.dir/SlowConverter.cc.o
[ 87%] Building CXX object CMakeFiles/fits2idia.dir/Util.cc.o
[100%] Linking CXX executable fits2idia
[100%] Built target fits2idia
```

--- a/src/Converter.h
+++ b/src/Converter.h
@@ -24,7 +24,7 @@ class Converter {
 public:
     Converter() {}
     Converter(std::string inputFileName, std::string outputFileName, bool progress);
-    ~Converter();
+    virtual ~Converter();
     
     static std::unique_ptr<Converter> getConverter(std::string inputFileName, std::string outputFileName, bool slow, bool progress);
     void convert();

